src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Symbols/SymbolTable.cs

```diff
@@ -46,6 +46,13 @@ public void InsertChild(ParentSymbol parent, Symbol child)
 
         private void InsertChildNoGrow(Symbol child)
         {
+            switch (child.getKind())
+            {
+                case SYMKIND.SK_Scope:
+                case SYMKIND.SK_LocalVariableSymbol:
```

Comment:
There might be more symbol kinds we can safely add to this list, and each added would give a slight performance bonus as well as perhaps guarding against a variant of the memory leak this fixes, but for now I'm just adding these two.